app/src/main/java/nextstep/jwp/controller/RegisterController.java

```diff
@@ -0,0 +1,34 @@
+package nextstep.jwp.controller;
+
+import nextstep.jwp.db.InMemoryUserRepository;
+import nextstep.jwp.domain.*;
+import nextstep.jwp.domain.request.HttpRequest;
+import nextstep.jwp.domain.request.RequestBody;
+import nextstep.jwp.domain.response.HttpResponse;
+import nextstep.jwp.domain.response.HttpStatus;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public class RegisterController extends AbstractController {
+
+    @Override
+    protected HttpResponse doGet(HttpRequest request) throws IOException {
+        HttpResponse response = new HttpResponse();
+        return response.respond(request.getUri() + ".html");
+    }
+
+    @Override
+    protected HttpResponse doPost(HttpRequest request) {
+        RequestBody requestBody = request.getRequestBody();
+        HttpResponse response = new HttpResponse();
+        Optional<User> user = InMemoryUserRepository.findByAccount(requestBody.getParam("account"));
+        if (user.isPresent()) {
+            return response.redirect("/400.html", HttpStatus.BAD_REQUEST);
+        }
+        int latest = InMemoryUserRepository.getLatestId();
+        User signupUser = new User(latest++, requestBody.getParam("account"), requestBody.getParam("password"), requestBody.getParam("email"));
```

Comment:
해당 부분에서 가입 시에 기존에 있던 1과 gugu가 중복되는 것 같습니다.
```suggestion
        int latest = InMemoryUserRepository.getLatestId();
        User signupUser = new User(++latest, requestBody.getParam("account"), requestBody.getParam("password"), requestBody.getParam("email"));
```
로 처리를 하거나, Repository에서 id에 +1을 해서 반환해줘야할 것 같은데 욘은 어떻게 생각하시는지 궁금합니다!

Reply:
AtomicLong으로 로직을 수정해봤습니다!